### DISCUSSION TITLE
20220531

### DISCUSSION THREAD

**Discussion by tmiw** [ARSFI/paclink-open-source#26] [General]
This is a pre-production build intended to gauge current status of the project and determine what needs to be fixed. Brief testing indicates that packet (using Kenwood TM-D700A), AGW (using AGWPE and UZ7HO soundmodem) and Telnet channels work; Pactor is TBD.

Contains (vs. [last pre-release version](https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20220526)):

* Pactor: add synchronization for outbound queues to prevent threading issues.

Instructions:

1. Unzip paclink-open-source.zip somewhere on your hard drive.
2. Run setup.exe to install on your system. Paclink will be installed in the Start menu.
3. Open Paclink from the Start menu and reconfigure if required. (There is no import functionality from the existing Paclink at this time but if you used a pre-release version, configuration should be saved.)

Please leave comments in the release's discussion thread or email K6AQ@winlink.org with feedback. 👍
 

<hr /><em>This discussion was created from the release <a href='https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20220531'>20220531</a>.</em>

**Comment by kd4vrf** [comment]
Sorry for the long delay, I tested positive for covid and had to self isolate. That made it difficult to get to the shack and do some testing.

I haven't seen the "Paclink has stopped working", I have seen the following with about the same frequency since the update:
![Screen Shot 2022-06-01 at 3 35 37 PM](https://user-images.githubusercontent.com/45251519/173385599-8b54f7ff-f5ee-404c-96a5-fc2277c2bd60.png)

The "Initialization failed on serial port COM7" error might be on my side. I've seen it a few times. I'm trying to rule out my setup:

Windows 10 VM --> Windows serial port to bluetooth emulator --> VM hardware abstraction --> physical bluetooth adapter on mac mini (tty.SCSDRAGON7800-SerialPort) --> Bluetooth --> Pactor DR-7800.

When I see that error i'm able to use "screen /dev/tty.SCSDRAGON7800-SerialPort" to open a terminal to the DR-7800 directly so it appears to be functioning from "physical bluetooth adapter on mac mini (tty.SCSDRAGON7800-SerialPort) --> Bluetooth --> Pactor DR-7800." when I see the error. I haven't been able to determine why the Windows VM thinks COM7 is open, everything I can find online to track down what has the serial port open in Windows doesn't apply for a bluetooth serial emulated port. So I haven't been able to determine if it's my setup or Paclink that isn't closing the port properly. The only thing I run Windows for is Winlink so I'm not as adept at troubleshooting Windows.

I do see this error in Event Viewer every time I get the COM7 error:
![Screen Shot 2022-06-13 at 11 24 21 AM](https://user-images.githubusercontent.com/45251519/173388834-5c535c12-6a6a-49e6-ac50-30a4ebf76daf.png)

I've updated my VMware Tools, VMware, Windows, and checked to make sure I have the latest drivers for the DR-7800 but I haven't had a chance to see if any of those steps resolved the issue.

When I get into the failed state I'll keep trying to rule out my setup, it's got more moving parts than the average users setup because of my Windows VM. This issue might be unique to my setup and not something that others will run into.

I was able to test the "off by one indexing issue" with the large number of addresses in the "To:" field. I was able to retrieve the message that has 400+ recipients from the Winlink Wednesday net and it worked as expected.

**Reply by tmiw** [reply]
Glad that you're feeling better and that at least the large email list issue has been resolved. 👍 

Re: Bluetooth, it sounds like you may be sharing the serial port device that macOS creates for the DR-7800. In the version of VMware I run, it seems like it might be possible to share the Bluetooth device directly based on this option I saw on one of my VMs:

<img width="621" alt="Screen Shot 2022-06-13 at 7 59 49 PM" src="https://user-images.githubusercontent.com/449242/173484445-0af881ea-e227-463f-a9a3-83b008ab1bc1.png">

I wonder if that would potentially work better if you can use that on Windows. I'd also try to use a USB cable instead of BT (and connect the USB device to the VM) to rule out BT issues. Let me know once you find out more about the issue for sure. Unfortunately this might be difficult to resolve here if it is indeed something with the way Paclink handles Bluetooth devices (I have a PTC-IIex but of course you need a physical serial cable for that).

**Reply by kd4vrf** [reply]
In my MacOS VMware I have that same dialog box and that is how I am sharing the Mac mini bluetooth with the VM. It does recognize the DR-7800 bluetooth and it's connected to Windows.

I did switch to a USB cable instead and I haven't seen the same disconnect, based on that I think it's probably Windows / VM related and outside the scope of Paclink.

I did notice two other issues:

When you go to change the Frequency for an RMS Gateway that has multilple frequencies available it lists all of the available frequencies twice:
![Screen Shot 2022-06-14 at 9 01 32 PM](https://user-images.githubusercontent.com/45251519/173714321-2d413404-9f1b-4540-bfba-1d639f13e6cd.png)

Also, no matter what frequency you select the first time it reverts back to whatever frequency it was on when you first click the drop down:

https://user-images.githubusercontent.com/45251519/173714716-b3f5b6cd-d682-411a-b9bc-4acec4e00685.mov

You need to select the frequency twice to select a different frequency.

**Reply by tmiw** [reply]
I noticed some weird behavior in the Pactor settings as well (see https://github.com/ARSFI/paclink-open-source/issues/27). I'm going to get some clarification first before I fix both issues so I don't accidentally introduce another bug.

**Reply by tmiw** [reply]
I think I resolved at least some of the above with the build I just made: https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20220617. Let me know if you run into anything else!
